"Slice bounds out of range" error with the Unity segment

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

When I go to a folder with one of my Unity projects in it..

```oh-my-posh fatal error rendering unity segment:runtime error: slice bounds out of range [:-1]

goroutine 25 [running]:
runtime/debug.Stack()
        /opt/hostedtoolcache/go/src/runtime/debug/stack.go:24 +0x65
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetEnabled.func1()
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:466 +0x58
panic({0xe82fc0, 0xc000794000})
        /opt/hostedtoolcache/go/src/runtime/panic.go:884 +0x213
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Unity).GetUnityVersion(0xc0003d2200)
        /home/runner/work/oh-my-posh/oh-my-posh/src/segments/unity.go:71 +0x2d0
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Unity).Enabled(0xc0003d2200)
        /home/runner/work/oh-my-posh/oh-my-posh/src/segments/unity.go:23 +0x25
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetEnabled(0xc0003de340, {0x1043068, 0xc000289ab0})
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:486 +0x177
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setEnabledSegments.func1(0x0?)
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:112 +0x6a
created by github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).setEnabledSegments
        /home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:110 +0x98
```

This seems to happen for any Unity project for me.


### Theme

I don't know what I derived mine from, but here's my prompt config:

[prompt.zip](https://github.com/JanDeDobbeleer/oh-my-posh/files/11222648/prompt.zip)


### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
oh-my-posh fatal error rendering unity segment:runtime error: slice bounds out of range [:-1]

goroutine 1 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/src/runtime/debug/stack.go:24 +0x65
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetEnabled.func1()
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:466 +0x58
panic({0xe82fc0, 0xc00041f1a0})
	/opt/hostedtoolcache/go/src/runtime/panic.go:884 +0x213
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Unity).GetUnityVersion(0xc0003e65c0)
	/home/runner/work/oh-my-posh/oh-my-posh/src/segments/unity.go:71 +0x2d0
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Unity).Enabled(0xc0003e65c0)
	/home/runner/work/oh-my-posh/oh-my-posh/src/segments/unity.go:23 +0x25
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Segment).SetEnabled(0xc000298d00, {0x1043068, 0xc00028da40})
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/segment.go:486 +0x177
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Block).Debug(0xc0001808f0)
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/block.go:232 +0x1cf
github.com/jandedobbeleer/oh-my-posh/src/engine.(*Engine).PrintDebug(0xc0000c9cd8, {0xecd815?, 0x0?, 0x1ac0380?}, {0x102cfd0, 0x7})
	/home/runner/work/oh-my-posh/oh-my-posh/src/engine/engine.go:283 +0xb72
github.com/jandedobbeleer/oh-my-posh/src/cli.glob..func7(0x1a97320?, {0xc000296520?, 0x1?, 0x1?})
	/home/runner/work/oh-my-posh/oh-my-posh/src/cli/debug.go:50 +0x327
github.com/spf13/cobra.(*Command).execute(0x1a97320, {0xc000296500, 0x1, 0x1})
	/opt/hostedtoolcache/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x1a98180)
	/opt/hostedtoolcache/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/opt/hostedtoolcache/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/jandedobbeleer/oh-my-posh/src/cli.Execute({0x102cfd0?, 0xc0000a0000?})
	/home/runner/work/oh-my-posh/oh-my-posh/src/cli/root.go:41 +0x4a
main.main()
	/home/runner/work/oh-my-posh/oh-my-posh/src/main.go:10 +0x27



Version: 14.29.1

Shell: oh-my-posh (7.2.10)

Segments:

ConsoleTitle(true)                         -   0 ms - C:\UnitySrc\work\_tests\empty-nsi :: oh-my-posh
os(true)                                   -   2 ms - εÿ¬ 
session(true)                              -   0 ms - scobeast
root(false)                                -   0 ms - 
path(true)                                 -   2 ms - εé░ C:/UnitySrc/work/_tests/empty-nsi 
unity(true)                                -   1 ms - εé░ ε£í  
git(true)                                  - 353 ms - εé░ ∩¼½ devΓëó ∩üä ~4
plastic(false)                             -  21 ms - 
exit(false)                                -   0 ms - 
executiontime(false)                       -   0 ms - 
time(true)                                 -   0 ms -  15:33:08
battery(false)                             -   6 ms - 
text(true)                                 -   0 ms - $

Run duration: 409.2004ms

Cache path: C:\Users\scott\AppData\Local\oh-my-posh

Config path: C:\Users\scott\.config\oh-my-posh\prompt.json

Logs:

[DEBUG] 15:33:08.072 shell.go:Getenv:398 ΓåÆ NO DATA
[TRACE] 15:33:08.072 shell.go:Getenv(TERM_PROGRAM) - 603.8┬╡s
[DEBUG] 15:33:08.072 shell.go:Getenv:398 ΓåÆ C:\Users\scott\AppData\Local
[TRACE] 15:33:08.072 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 15:33:08.072 shell_windows.go:CachePath() - 0s
[DEBUG] 15:33:08.073 shell.go:Getenv:398 ΓåÆ C:\Users\scott\.config\oh-my-posh\prompt.json
[TRACE] 15:33:08.073 shell.go:Getenv(POSH_THEME) - 0s
[DEBUG] 15:33:08.089 shell.go:Shell:645 ΓåÆ process name: oh-my-posh.exe
[TRACE] 15:33:08.089 shell.go:Shell() - 16.0847ms
[TRACE] 15:33:08.089 shell.go:resolveConfigPath() - 16.0847ms
[TRACE] 15:33:08.089 shell.go:Init() - 17.7231ms
[TRACE] 15:33:08.089 shell.go:Flags() - 0s
[TRACE] 15:33:08.090 config.go:loadConfig() - 1.1693ms
[TRACE] 15:33:08.090 shell.go:Flags() - 0s
[DEBUG] 15:33:08.090 shell.go:Getenv:398 ΓåÆ NO DATA
[TRACE] 15:33:08.090 shell.go:Getenv(OMP_CACHE_DISABLED) - 0s
[TRACE] 15:33:08.090 shell_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.090 shell_windows.go:WindowsRegistryKeyValue:206 ΓåÆ ColorizationColor(DWORD): 0xC4570F61
[TRACE] 15:33:08.090 shell.go:Shell() - 0s
[DEBUG] 15:33:08.090 shell.go:Getenv:398 ΓåÆ 7.2.10
[TRACE] 15:33:08.090 shell.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 15:33:08.090 engine.go:PrintDebug:266 ΓåÆ Segment: Title
[TRACE] 15:33:08.090 shell_windows.go:Root() - 0s
[TRACE] 15:33:08.090 shell.go:Shell() - 0s
[TRACE] 15:33:08.090 shell.go:ErrorCode() - 0s
[TRACE] 15:33:08.090 shell_windows.go:IsWsl() - 0s
[TRACE] 15:33:08.090 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.090 shell.go:Pwd:428 ΓåÆ C:\UnitySrc\work\_tests\empty-nsi
[TRACE] 15:33:08.090 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.090 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.091 shell.go:User:549 ΓåÆ scott
[TRACE] 15:33:08.091 shell.go:User() - 0s
[DEBUG] 15:33:08.091 shell.go:Host:561 ΓåÆ scobeast
[TRACE] 15:33:08.091 shell.go:Host() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.091 shell.go:Getenv:398 ΓåÆ NO DATA
[TRACE] 15:33:08.091 shell.go:Getenv(SHLVL) - 0s
[TRACE] 15:33:08.091 shell.go:TemplateCache() - 520.4┬╡s
[TRACE] 15:33:08.091 shell.go:Flags() - 0s
[DEBUG] 15:33:08.091 block.go:Debug:230 ΓåÆ Segment: os
[TRACE] 15:33:08.091 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.091 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[TRACE] 15:33:08.091 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.091 properties.go:GetString:28 ΓåÆ εÿ¬
[TRACE] 15:33:08.091 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.093 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.093 shell.go:Shell() - 0s
[DEBUG] 15:33:08.094 block.go:Debug:230 ΓåÆ Segment: session
[TRACE] 15:33:08.094 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.094 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.094 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.094 shell.go:Getenv:398 ΓåÆ NO DATA
[TRACE] 15:33:08.094 shell.go:Getenv(SSH_CONNECTION) - 0s
[DEBUG] 15:33:08.094 shell.go:Getenv:398 ΓåÆ NO DATA
[TRACE] 15:33:08.094 shell.go:Getenv(SSH_CLIENT) - 0s
[TRACE] 15:33:08.094 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.094 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.094 shell.go:Shell() - 0s
[DEBUG] 15:33:08.094 block.go:Debug:230 ΓåÆ Segment: root
[TRACE] 15:33:08.094 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.094 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.094 shell.go:GOOS() - 0s
[TRACE] 15:33:08.094 shell_windows.go:Root() - 0s
[DEBUG] 15:33:08.094 block.go:Debug:230 ΓåÆ Segment: path
[TRACE] 15:33:08.094 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.094 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.094 shell.go:GOOS() - 0s
[TRACE] 15:33:08.094 shell.go:Pwd() - 0s
[TRACE] 15:33:08.094 shell.go:Shell() - 0s
[TRACE] 15:33:08.095 shell.go:Shell() - 0s
[DEBUG] 15:33:08.095 properties.go:GetBool:22 ΓåÆ mapped_locations_enabled: true
[DEBUG] 15:33:08.095 properties.go:GetString:28 ΓåÆ ∩Çô
[DEBUG] 15:33:08.095 properties.go:GetString:28 ΓåÆ ∩Çô
[DEBUG] 15:33:08.095 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.095 properties.go:GetString:28 ΓåÆ ~
[DEBUG] 15:33:08.095 properties.go:GetKeyValueMap:46 ΓåÆ mapped_locations: map[]
[TRACE] 15:33:08.095 shell.go:PathSeparator() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:PathSeparator() - 0s
[TRACE] 15:33:08.095 shell.go:PathSeparator() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:PathSeparator() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:PathSeparator() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:GOOS() - 0s
[TRACE] 15:33:08.095 shell.go:PathSeparator() - 0s
[DEBUG] 15:33:08.095 properties.go:GetString:28 ΓåÆ full
[TRACE] 15:33:08.095 shell.go:PathSeparator() - 0s
[TRACE] 15:33:08.095 shell.go:PathSeparator() - 0s
[DEBUG] 15:33:08.095 properties.go:GetString:28 ΓåÆ NO DATA
[TRACE] 15:33:08.095 shell.go:PathSeparator() - 0s
[DEBUG] 15:33:08.095 properties.go:GetString:28 ΓåÆ /
[TRACE] 15:33:08.095 shell.go:Pwd() - 0s
[TRACE] 15:33:08.095 shell_windows.go:IsWsl() - 0s
[TRACE] 15:33:08.095 shell.go:StackCount() - 0s
[DEBUG] 15:33:08.096 win32_windows.go:isWriteable:303 ΓåÆ current user is member of S-1-5-32-544
[DEBUG] 15:33:08.096 win32_windows.go:isWriteable:311 Γåô
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 15:33:08.096 win32_windows.go:isWriteable:313 ΓåÆ user has write access
[TRACE] 15:33:08.096 shell_windows.go:DirIsWritable() - 1.5901ms
[TRACE] 15:33:08.096 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.097 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.097 shell.go:Shell() - 0s
[DEBUG] 15:33:08.097 block.go:Debug:230 ΓåÆ Segment: unity
[TRACE] 15:33:08.097 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.097 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.097 shell.go:GOOS() - 0s
[TRACE] 15:33:08.097 shell.go:Pwd() - 0s
[TRACE] 15:33:08.097 shell.go:HasParentFilePath(ProjectSettings) - 0s
[DEBUG] 15:33:08.097 shell.go:HasFilesInDir:464 ΓåÆ true
[TRACE] 15:33:08.097 shell.go:HasFilesInDir(ProjectVersion.txt) - 0s
[DEBUG] 15:33:08.097 shell.go:FileContent:523 Γåô
    m_EditorVersion: 2023.2.0a9
    m_EditorVersionWithRevision: 2023.2.0a9 (5405d0db74a0)
    
[TRACE] 15:33:08.097 shell.go:FileContent(C:\UnitySrc\work\_tests\empty-nsi\ProjectSettings\ProjectVersion.txt) - 530.8┬╡s
[TRACE] 15:33:08.098 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.098 shell.go:Shell() - 0s
[DEBUG] 15:33:08.098 block.go:Debug:230 ΓåÆ Segment: git
[TRACE] 15:33:08.098 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.098 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.098 shell.go:GOOS() - 0s
[TRACE] 15:33:08.098 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.221 shell.go:CommandPath:600 ΓåÆ C:\Users\scott\scoop\shims\git.exe
[TRACE] 15:33:08.221 shell.go:CommandPath(git.exe) - 123.1501ms
[TRACE] 15:33:08.221 shell.go:HasCommand(git.exe) - 123.1501ms
[TRACE] 15:33:08.221 shell.go:Pwd() - 0s
[TRACE] 15:33:08.221 shell.go:HasParentFilePath(.git) - 0s
[DEBUG] 15:33:08.221 properties.go:GetStringArray:52 ΓåÆ exclude_folders: []
[DEBUG] 15:33:08.221 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.221 shell.go:Getenv:398 ΓåÆ NO DATA
[TRACE] 15:33:08.221 shell.go:Getenv(POSH_GIT_STATUS) - 0s
[DEBUG] 15:33:08.221 properties.go:GetBool:22 ΓåÆ fetch_status: true
[DEBUG] 15:33:08.221 properties.go:GetStringArray:52 ΓåÆ ignore_status: []
[DEBUG] 15:33:08.221 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.221 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.221 properties.go:GetKeyValueMap:46 ΓåÆ untracked_modes: map[C:/UnitySrc/work/unity/:no C:/UnitySrc/work/unity2/:no C:/UnitySrc/work/unity3/:no C:/UnitySrc/work/unity4/:no C:/UnitySrc/work/unity5/:no D:/UnitySrc/work/unity/:no D:/UnitySrc/work/unity2/:no D:/UnitySrc/work/unity3/:no D:/UnitySrc/work/unity4/:no D:/UnitySrc/work/unity5/:no]
[DEBUG] 15:33:08.221 properties.go:GetKeyValueMap:46 ΓåÆ ignore_submodules: map[]
[DEBUG] 15:33:08.343 shell.go:RunCommand:579 Γåô
    # branch.oid 65a910c17a5091e4aff483123eedce40fd56318b
    # branch.head dev
    1 .M N... 100644 100644 100644 5290acdeb02aba54c8b42513eca310b860e85e6c 5290acdeb02aba54c8b42513eca310b860e85e6c Packages/manifest.json
    1 .M N... 100644 100644 100644 66e1a43d1d686dd1ccccf2d26164a1a99e138b20 66e1a43d1d686dd1ccccf2d26164a1a99e138b20 Packages/packages-lock.json
    1 .M N... 100644 100644 100644 3e2cc8049ed6d44826711c1de6abc56b4b55311a 3e2cc8049ed6d44826711c1de6abc56b4b55311a ProjectSettings/ProjectSettings.asset
    1 .M N... 100644 100644 100644 0bfcec0ba7f99a7f8bf22745b86228c5ca3e51ae 0bfcec0ba7f99a7f8bf22745b86228c5ca3e51ae ProjectSettings/ProjectVersion.txt
[TRACE] 15:33:08.343 shell.go:RunCommand(git.exe -C C:/UnitySrc/work/_tests/empty-nsi --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --branch --porcelain=2) - 121.7667ms
[DEBUG] 15:33:08.343 properties.go:GetString:28 ΓåÆ ∩¼½ 
[DEBUG] 15:33:08.343 properties.go:GetInt:40 ΓåÆ branch_max_length: 0
[DEBUG] 15:33:08.343 shell.go:HasFolder:493 ΓåÆ false
[TRACE] 15:33:08.343 shell.go:HasFolder(C:\UnitySrc\work\_tests\empty-nsi\.git/rebase-merge) - 0s
[DEBUG] 15:33:08.343 shell.go:HasFolder:493 ΓåÆ false
[TRACE] 15:33:08.343 shell.go:HasFolder(C:\UnitySrc\work\_tests\empty-nsi\.git/rebase-apply) - 507.1┬╡s
[DEBUG] 15:33:08.343 properties.go:GetString:28 ΓåÆ ∩Éù
[DEBUG] 15:33:08.343 shell.go:HasFilesInDir:464 ΓåÆ false
[TRACE] 15:33:08.343 shell.go:HasFilesInDir(MERGE_MSG) - 0s
[DEBUG] 15:33:08.344 shell.go:HasFilesInDir:464 ΓåÆ false
[TRACE] 15:33:08.344 shell.go:HasFilesInDir(CHERRY_PICK_HEAD) - 524.9┬╡s
[DEBUG] 15:33:08.344 shell.go:HasFilesInDir:464 ΓåÆ false
[TRACE] 15:33:08.344 shell.go:HasFilesInDir(REVERT_HEAD) - 0s
[DEBUG] 15:33:08.344 shell.go:HasFilesInDir:464 ΓåÆ false
[TRACE] 15:33:08.344 shell.go:HasFilesInDir(sequencer/todo) - 0s
[DEBUG] 15:33:08.344 properties.go:GetString:28 ΓåÆ Γëó
[DEBUG] 15:33:08.344 properties.go:GetBool:22 ΓåÆ fetch_upstream_icon: true
[ERROR] 15:33:08.449 shell.go:RunCommand:577 ΓåÆ exit status 2
[DEBUG] 15:33:08.449 shell.go:RunCommand:579 Γåô
    error: No such remote 'origin'
    
[TRACE] 15:33:08.449 shell.go:RunCommand(git.exe -C C:/UnitySrc/work/_tests/empty-nsi --no-optional-locks -c core.quotepath=false -c color.status=false remote get-url origin) - 104.9088ms
[TRACE] 15:33:08.449 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.450 shell.go:TemplateCache() - 0s
[ERROR] 15:33:08.450 shell.go:FileContent:519 ΓåÆ open C:\UnitySrc\work\_tests\empty-nsi\.git/logs/refs/stash: The system cannot find the file specified.
[TRACE] 15:33:08.450 shell.go:FileContent(C:\UnitySrc\work\_tests\empty-nsi\.git/logs/refs/stash) - 586.2┬╡s
[DEBUG] 15:33:08.451 shell.go:HasFolder:493 ΓåÆ false
[TRACE] 15:33:08.451 shell.go:HasFolder(C:\UnitySrc\work\_tests\empty-nsi\.git/worktrees) - 538.4┬╡s
[TRACE] 15:33:08.451 shell.go:Shell() - 0s
[DEBUG] 15:33:08.451 block.go:Debug:230 ΓåÆ Segment: plastic
[TRACE] 15:33:08.451 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.451 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.451 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.471 shell.go:CommandPath:600 ΓåÆ C:\Program Files\PlasticSCM5\client\cm.exe
[TRACE] 15:33:08.471 shell.go:CommandPath(cm) - 20.3779ms
[TRACE] 15:33:08.471 shell.go:HasCommand(cm) - 20.3779ms
[TRACE] 15:33:08.471 shell.go:Pwd() - 0s
[ERROR] 15:33:08.472 shell.go:HasParentFilePath:733 ΓåÆ CreateFile .plastic: The system cannot find the file specified.
[TRACE] 15:33:08.472 shell.go:HasParentFilePath(.plastic) - 1.0342ms
[DEBUG] 15:33:08.472 block.go:Debug:230 ΓåÆ Segment: exit
[TRACE] 15:33:08.472 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.472 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.472 shell.go:GOOS() - 0s
[TRACE] 15:33:08.472 shell.go:ErrorCode() - 0s
[DEBUG] 15:33:08.472 properties.go:GetBool:22 ΓåÆ always_enabled: false
[TRACE] 15:33:08.472 shell.go:ErrorCode() - 0s
[TRACE] 15:33:08.472 shell.go:Flags() - 0s
[DEBUG] 15:33:08.472 block.go:Debug:230 ΓåÆ Segment: executiontime
[TRACE] 15:33:08.472 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.472 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.472 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.472 properties.go:GetBool:22 ΓåÆ always_enabled: false
[TRACE] 15:33:08.472 shell.go:ExecutionTime() - 0s
[DEBUG] 15:33:08.472 properties.go:GetFloat64:34 ΓåÆ threshold: 2000.000000
[DEBUG] 15:33:08.472 block.go:Debug:230 ΓåÆ Segment: time
[TRACE] 15:33:08.472 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.472 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.472 shell.go:GOOS() - 0s
[DEBUG] 15:33:08.472 properties.go:GetString:28 ΓåÆ 15:04:05
[TRACE] 15:33:08.472 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.473 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.473 shell.go:Shell() - 0s
[DEBUG] 15:33:08.473 block.go:Debug:230 ΓåÆ Segment: battery
[TRACE] 15:33:08.473 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.473 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.473 shell.go:GOOS() - 0s
[TRACE] 15:33:08.473 shell_windows.go:IsWsl() - 0s
[ERROR] 15:33:08.480 shell_windows_nix.go:BatteryState:15 ΓåÆ no battery
[TRACE] 15:33:08.480 shell_windows_nix.go:BatteryState() - 6.7873ms
[TRACE] 15:33:08.480 shell.go:Flags() - 0s
[DEBUG] 15:33:08.480 block.go:Debug:230 ΓåÆ Segment: text
[TRACE] 15:33:08.480 shell.go:Pwd() - 0s
[DEBUG] 15:33:08.480 shell_windows.go:56 ΓåÆ C:\Users\scott
[TRACE] 15:33:08.480 shell.go:GOOS() - 0s
[TRACE] 15:33:08.480 shell.go:TemplateCache() - 0s
[TRACE] 15:33:08.480 shell.go:Shell() - 0s
[DEBUG] 15:33:08.480 shell.go:Getenv:398 ΓåÆ C:\Users\scott\AppData\Local
[TRACE] 15:33:08.480 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 15:33:08.480 shell_windows.go:CachePath() - 0s
[TRACE] 15:33:08.480 shell.go:Flags() - 0s
```

--- a/src/segments/unity.go
+++ b/src/segments/unity.go
@@ -54,24 +54,25 @@ func (u *Unity) GetUnityVersion() (version string, err error) {
 	versionFilePath := filepath.Join(projectDir.Path, "ProjectVersion.txt")
 	versionFileText := u.env.FileContent(versionFilePath)
 
-	firstLine := strings.Split(versionFileText, "\n")[0]
+	lines := strings.Split(versionFileText, "\n")
 	versionPrefix := "m_EditorVersion: "
-	versionPrefixIndex := strings.Index(firstLine, versionPrefix)
-	if versionPrefixIndex == -1 {
-		err := errors.New("ProjectSettings/ProjectVersion.txt is missing 'm_EditorVersion: ' prefix")
-		return "", err
+	for _, line := range lines {
+		if !strings.HasPrefix(line, versionPrefix) {
+			continue
+		}
+		version := strings.TrimPrefix(line, versionPrefix)
+		version = strings.TrimSpace(version)
+		if len(version) == 0 {
+			return "", errors.New("Empty m_EditorVersion")
+		}
+		fIndex := strings.Index(version, "f")
+		if fIndex > 0 {
+			return version[:fIndex], nil
+		}
+		return version, nil
 	}
 
-	versionStartIndex := versionPrefixIndex + len(versionPrefix)
-	unityVersion := firstLine[versionStartIndex:]
-	unityVersion = strings.TrimSpace(unityVersion)
-
-	fIndex := strings.Index(unityVersion, "f")
-	if versionPrefixIndex > -1 {
-		unityVersion = unityVersion[:fIndex]
-	}
-
-	return unityVersion, nil
+	return "", errors.New("ProjectSettings/ProjectVersion.txt is missing m_EditorVersion")
 }
 
 func (u *Unity) GetCSharpVersion() (version string, err error) {
@@ -103,6 +104,7 @@ func (u *Unity) GetCSharpVersion() (version string, err error) {
 		"2022.1": "C# 9",
 		"2022.2": "C# 9",
 		"2023.1": "C# 9",
+		"2023.2": "C# 9",
 	}
 
 	csharpVersion, found := csharpVersionsByUnityVersion[shortUnityVersion]

--- a/src/segments/unity_test.go
+++ b/src/segments/unity_test.go
@@ -40,6 +40,13 @@ func TestUnitySegment(t *testing.T) {
 		VersionFileExists   bool
 	}{
 		{
+			Case:                "Unity version without f suffix",
+			ExpectedOutput:      "\ue721 2023.2.0a9 C# 9",
+			ExpectedToBeEnabled: true,
+			VersionFileExists:   true,
+			VersionFileText:     "m_EditorVersion: 2023.2.0a9\nm_EditorVersionWithRevision: 2023.2.0a9 (5405d0db74a0)",
+		},
+		{
 			Case:                "Unity version exists in C# map",
 			ExpectedOutput:      "\ue721 2021.3.16 C# 9",
 			ExpectedToBeEnabled: true,

